HIP/hip-0000-cancun-4.md

```diff
@@ -0,0 +1,163 @@
+make future upgrades achievable. The removal of all contract state as part of
+self-destruction would make future data model changes prohibitively expensive.
+However, there were a number of contracts that depended on side effects of
+SELFDESTRUCT not related to the contract storage and contract code removal,
+namely the transfer of Ether to a beneficiary address. Similarly, some DeFi
+usage patterns used "transient" contracts to store data between calls within the
+same transaction, and self-destruction was key to being a good citizen and not
+leaving garbage around.
+
+To reduce the damage that would occur from simply disabling the opcode Ethereum
+decided to change the behavior of SELFDESTRUCT depending on the age of the
+contract. If the contract existed prior to the start of the transaction the
+operation would change so that neither the code nor the storage was deleted, but
+the ether balance would be sent to the beneficiary. This moved the operation
+into a "sweep" operation that allowed contracts to clean up all balance without
+having to query it. For contracts that did not exist prior to the transaction
+SELFDESTRUCT would continue to operate normally. Balance would be sent to the
+beneficiary, and code and storage would be deleted. Since none of this data had
+been part of the committed data the deletion did not impact any plans for
+changes to their storage model.
+
+Within the EVM Hedera should support a similar model, as contracts will now
+expect SELFDESTRUCT to operate as a sweep in some cases. Outside the EVM,
+however, there is no need to change any semantics. HAPI calls can still be used
+to delete a contract and state expiry can still delete an account.
+
+## Rationale
+
+Hedera will support the [EIP-6780](https://eips.ethereum.org/EIPS/eip-6780)
+change to the SELFDESTRUCT operation. If the contract existed prior to the
+transaction the contract will not be destroyed and the contents will instead be
+swept to the beneficiary. Contracts within the same transaction will be
+destroyed as per prior behavior.
+
+However, since Hedera places "ownership" of token balances with the account
+holding the balance we will need to extend the behavior of the sweep mode to
+also transfer all HTS tokens to the beneficiary. This is consistent with current
+self-destruct behavior and consistent with the spirit of the Ethereum change in
+that code and the account are no longer deleted.
+
+When the self-destruct is invoked on a contract within a transaction the
+behavior of Hedere will remain as it currently is in all respects.
+
+## User stories
+
+* As a smart contract developer I want to use the "sweep" semantics of EIP-6780
+  in my contracts to sweep out hbar while not destroying the contract.
+* As a smart contract developer I want to be able to continue to use the
+  transient contract pattern and have it behave the same way prior to this HIP.
+* As a HAPI user I want to be able to control my contract account in the same
```

Comment:
Not sure this is an actual user story. Should be clear what flow should be possible or not possible